### PR TITLE
Release File Update for PT1.12.1

### PR DESCRIPTION
**Pull request by ohadkatz**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .release_images_template.yml [hunk 1]
```diff
@@ -966,3 +966,77 @@ release_images:
       # to images being published.
       force_release: False   # [Default: False] Set to True to force images to be published even if the same image
       # has already been published. Re-released image will have minor version incremented by 1.
+  62:
+    framework: "pytorch"
+    version: "1.12.1"
```
Comment (tejaschumbalkar) [inline .release_images_template.yml:971]: Update the version of `v1.12.0` on 59 and 60 entry instead of adding a new one

File: .release_images_template.yml [hunk 2]
```diff
@@ -966,3 +966,77 @@ release_images:
...
+    arch_type: "x86"
+    training:
+      device_types: [ "gpu" ]
+      python_versions: [ "py38" ]
+      os_version: "ubuntu20.04"
+      cuda_version: "cu113"
+      example: False
+      disable_sm_tag: False  # [Default: False] This option is not used by Example images
+      force_release: False
+  65:
```
Comment (tejaschumbalkar) [inline .release_images_template.yml:1019]: config already exists, lets remove this entry

File: release_images.yml [hunk 3]
```diff
@@ -50,3 +50,28 @@ release_images:
       os_version: "ubuntu20.04"
       example: False
       disable_sm_tag: True
```
Comment (tejaschumbalkar) [inline release_images.yml:52]: remove all the entries above since these are added in the .release_images_template.yml file

File: .release_images_template.yml [hunk 4]
```diff
@@ -966,3 +966,38 @@ release_images:
       # to images being published.
       force_release: False   # [Default: False] Set to True to force images to be published even if the same image
       # has already been published. Re-released image will have minor version incremented by 1.
+  63:
```
Comment (tejaschumbalkar) [inline .release_images_template.yml:969]: nit: fix numbering starting from 61

File: release_images.yml [hunk 5]
```diff
@@ -3,50 +3,51 @@ release_images:
...
+      python_versions: [ "py38" ]
       os_version: "ubuntu20.04"
-      example: False
-      disable_sm_tag: True
+      cuda_version: "cu113"
+      example: False        # [Default: False] Set to True to denote that this image is an Example image
+      disable_sm_tag: False # [Default: False] Set to True to prevent SageMaker Abbreviated Tags from being attached
+      # to images being published.
+      force_release: False   # [Default: False] Set to True to force images to be published even if the same image
+      # has already been published. Re-released image will have minor version incremented by 1.
```
Comment (tejaschumbalkar) [inline release_images.yml:53]: Add configuration for example image as well like https://github.com/aws/deep-learning-containers/blob/master/.release_images_template.yml#L929

File: release_images.yml [hunk 6]
```diff
@@ -3,50 +3,51 @@ release_images:
...
     framework: "pytorch"
-    version: "1.5.1"
+    version: "1.12.1"
     arch_type: "x86"
+    customer_type: "ec2"
+    training:
+      device_types: [ "cpu", "gpu" ]
+      python_versions: [ "py38" ]
+      os_version: "ubuntu20.04"
+      cuda_version: "cu113"
```
Comment (tejaschumbalkar) [inline release_images.yml:38]: it should be `cu116`

File: .release_images_template.yml [hunk 7]
```diff
@@ -966,3 +966,38 @@ release_images:
       # to images being published.
       force_release: False   # [Default: False] Set to True to force images to be published even if the same image
       # has already been published. Re-released image will have minor version incremented by 1.
+  62:
```
Comment (tejaschumbalkar) [inline .release_images_template.yml:969]: this should be 61

File: release_images.yml [hunk 8]
```diff
@@ -3,50 +3,66 @@ release_images:
...
     framework: "pytorch"
-    version: "1.10.2"
+    version: "1.12.1"
     arch_type: "x86"
+    customer_type: "sagemaker"
+    training:
+      device_types: [ "cpu", "gpu" ]
+      python_versions: [ "py38" ]
+      os_version: "ubuntu20.04"
+      cuda_version: "cu116"
```
Comment (tejaschumbalkar) [inline release_images.yml:13]: cuda version should be `cu113` for SM and `cu116` for E3

Comment (tejaschumbalkar) [reply release_images.yml:13]: ref uri in https://github.com/aws/deep-learning-containers/blob/master/available_images.md

File: .release_images_template.yml [hunk 9]
```diff
@@ -920,15 +920,15 @@ release_images:
       device_types: [ "cpu", "gpu" ]
       python_versions: [ "py38" ]
       os_version: "ubuntu20.04"
-      cuda_version: "cu116"
+      cuda_version: "cu113"
```
Comment (tejaschumbalkar) [inline .release_images_template.yml:923]: this needs to be `cu116` since this is a EC2 inference configuration

File: release_images.yml (hunk 8 above)
Comment (tejaschumbalkar) [inline release_images.yml:13]: this needs to be `cu113` since this is a SM training configuration

**Review by tejaschumbalkar — DISMISSED** [review]

**Review by tejaschumbalkar — DISMISSED** [review]

**Review by tejaschumbalkar — APPROVED** [review]
